### ISSUE TITLE
Export Patent Applications as such

### ISSUE THREAD

**Issue by nils-werner** [retorquere/zotero-better-bibtex#1413]
**Report ID:** `ZF9DFM5I-euc` and `ZN27ABPT-euc`

**Exporter used:** Better BibLaTeX

**Expected behavior:**

BibLaTeX encodes an *EU patent* with number EU456456456 as

    @patent{key,
      [...]
      number = {456456456},
      type = {patenteu}
    }

but it also allows to encode an *EU patent application* with application number EU123123123 as

    @patent{key,
      [...]
      number = {123123123},
      type = {patreqeu}
    }

Same holds for other patent offices.

I would expect Better BibLaTeX to

 - If an *Application Number* EU123123123 is set, but **no *Patent Number*** is set
    - select `type = {patreqeu}`,
    - set `number = {123123123},` equal to the *Application Number*, minus the EU prefix
 - If a *Patent Number* EU456456456 is set,
    - select `type = {patenteu}`,
    - set `number = {456456456},` equal to the *Patent Number*, minus the EU prefix
    - disregard the *Application Number*, if it is there

**Actual behavior:**

Better BibLaTeX ignores *Application Number* and simply exports a `type = {patenteu}` with *Patent Number*.

**Comment by nils-werner** [comment]
Report IDs `ZF9DFM5I-euc` and `ZN27ABPT-euc` amended to the original post.

**Comment by nils-werner** [comment]
A workaround is to set

    tex.type: patreqeu
    tex.number: 123123123

in the *Extras* field in Zotero.

**Comment by retorquere** [comment]
It's a little hard to follow with the synthetic examples above. Can you show what you want based on the actual two items you sent?

**Comment by retorquere** [comment]
I really can't do anything without further feedback.

**Comment by retorquere** [comment]
Hello?

**Comment by retorquere** [comment]
I'm currently waiting for this and #1414 before I cut a new release with some major performance work. I really can't do anything without input from you. The log files that were sent do not relate to the examples above in any way that I can recognize.

**Comment by nils-werner** [comment]
I have created two example reports:

### Patent Example

`2WCSZHNZ-euc` is an *actual patent* with the Zotero data

| Field | Data |
|-------|------|
| Title | This Is a Patent. It Has the Patent Number EU123123123. It Had the Application Number EU789789789 When It Was First Submitted. |
| Author | Werner, Nils |
| Holder | My Employer |
| Patent Number | EP123123123 |
| Application Number | EP789789789 |

Note that it has a *Patent Number*. It may also optionally have an *Application Number*, which is not exported.

it exports correctly to

    @patent{wernerThisPatentIt,
      title = {This Is a Patent. It Has the Patent Number EP123123123. It Had the Application Number EP789789789 When It Was First Submitted.},
      author = {Werner, Nils},
      holder = {{My Employer}},
      number = {123123123},
      type = {patenteu}
    }

### Patent Application Example

`IIWRMX6L-euc` is a patent application

| Field | Data |
|-------|------|
| Title | This is a Patent Application. It has the Application Number EP456456456 |
| Author | Werner, Nils |
| Holder | My Employer |
| Application Number | EP456456456 |

Note it only has an *Application Number* but no *Patent Number*. This distinguishes it from an actual patent!

It exports incorectly to

    @patent{wernerThisPatentApplication,
      title = {This Is a Patent Application. It Has the Application Number EP456456456},
      author = {Werner, Nils},
      holder = {{My Employer}},
      type = {patent}
    }

Instead it should export to

    @patent{wernerThisPatentApplication,
      title = {This Is a Patent Application. It Has the Application Number EP456456456},
      author = {Werner, Nils},
      holder = {{My Employer}},
      number = {456456456},
      type = {patreqeu}
    }

**Comment by retorquere** [comment]
Cool -- that gives me a successfully failing test, which will also prevent future regressions. This shouldn't be hard to fix.

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.2.14.5807](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.14.5807.xpi) ("fixes #1413")

Install in Zotero by downloading [test build 5.2.14.5807](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.14.5807.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
try 5807

**Comment by nils-werner** [comment]
5807 works as expected, thanks!

**Comment by retorquere** [comment]
Super.

Only I can close issues here, I keep them open so I remember to merge them into the release.
